packages/react-checkbox/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -1,28 +1,25 @@
 import * as React from 'react';
-import { ComponentPropsCompat, ComponentStateCompat, ShorthandProps } from '@fluentui/react-utilities';
+import { ComponentProps, ComponentState } from '@fluentui/react-utilities';
 import { LabelProps } from '@fluentui/react-label';
 
-/**
- * Checkbox Props
- */
-export interface CheckboxProps
-  extends Omit<ComponentPropsCompat, 'children'>,
-    Omit<React.HTMLAttributes<HTMLElement>, 'defaultChecked' | 'onChange'> {
+export type CheckboxSlots = {
   /**
    * Label that will be rendered next to the checkbox.
    */
-  label?: ShorthandProps<LabelProps>;
+  // label: LabelProps;
 
   /**
-   * Indicator to be rendered as the checkbox icon.
+   * Hidden input that handles the checkbox's functionality.
    */
-  indicator?: ShorthandProps<ComponentPropsCompat>;
+  input: React.InputHTMLAttributes<HTMLInputElement> & React.RefAttributes<HTMLInputElement>;
 
   /**
-   * Hidden input that handles the checkbox's functionality.
+   * Indicator to be rendered as the checkbox icon.
    */
-  input?: ShorthandProps<React.InputHTMLAttributes<HTMLInputElement> & React.RefAttributes<HTMLInputElement>>;
+  indicator: React.HtmlHTMLAttributes<HTMLDivElement>;
+};
 
+export interface CheckboxCommons extends Omit<LabelProps, 'defaultChecked' | 'onChange' | 'as'> {
```

Comment:
1. Sounds good.
2. Yes, it's because commons shouldn't have the `as` prop.
3. `LabelProps` does have `defaultChecked` and `onChange`, so I need the omit otherwise it complains that the types are not compatible.